backend/authentication/tests.py

```diff
@@ -24,3 +28,106 @@ def test_str_methods() -> None:
     assert str(user_resource) == str(user_resource.id)
     assert str(user_task) == str(user_task.id)
     assert str(user_topic) == str(user_topic.id)
+
+
+@pytest.mark.django_db
+def test_signup(client: Client) -> None:
+    """Test the signup function.
+
+    Scenarios:
+
+    1. Password strength fails
+    2. Password confirmation fails
+    3. User is created successfully
+       - Check response status code
+       - Check if user exists in the DB
+       - Check if user password is hashed
+         (simple check if the password in the DB is the same as the raw password)
+    4. User already exists / Username already exists
+    5. Different User with the same email already exists
+
+    """
+    # Setup
+    fake = Faker()
+    username = fake.name()
+    second_username = fake.name()
+    email = fake.email()
+    strong_password = fake.password(
+        length=12, special_chars=True, digits=True, upper_case=True
+    )
+    wrong_password_confirmed = fake.password(
```

Comment:
Want to check really quick on Faker here as the percentage chance that the passwords generated would be the same when generated at random is ~0, but is there also something in Faker that within a given instance it's not going to return the same thing twice? Just wondering.

Reply:
Every time you invoke the method it will generate a random password. It is very unlikely to get the same password for our cases, but why would you not want the chance of the same password? Users also may also have the same password without knowing it, so do our tests 😄

Reply:
Just mean that for that for the tests in this case we'd want this to 100% not be `strong_password`, and it's 99.99999%, so all good, but was wondering if there was something that meant that on instantiation we'd definitely not generate the same thing twice. Thanks for the explanation!